README.md

```diff
@@ -1,8 +1,66 @@
-# sdg
+# Synthetic Data Generation (SDG)
 
 ![Lint](https://github.com/instructlab/sdg/actions/workflows/lint.yml/badge.svg?branch=main)
 ![Build](https://github.com/instructlab/sdg/actions/workflows/pypi.yaml/badge.svg?branch=main)
 ![Release](https://img.shields.io/github/v/release/instructlab/sdg)
 ![License](https://img.shields.io/github/license/instructlab/sdg)
 
 Python library for Synthetic Data Generation
+
+## Introduction 
+
+Synthetic Data Generation (SDG) is a process that creates an artificially generated dataset that mimics real data based on provided examples. SDG uses a YAML file containing question-and-answer pairs as input data.
+
+## Installing the SDG library
+
+Clone the library and navigate to the repo:
+```bash
+git clone https://github.com/instructlab/sdg
+cd sdg
+```
+
+Install the library:
+```bash
+pip install .
+```
+
+## Using the library 
+
+You can use the SDG library with the following items
+
+```bash
+ from instructlab.sdg.generate_data import generate_data
+ from instructlab.sdg.utils import GenerateException
+```
+
+<!--Not sure what more your thinking of adding here -->
+
+## Pipelines 
+
+There are four pipelines that are used in SDG. Each pipeline requires specific hardware specifications. 
+<!--TODO: Add explanations of pipelines-->
```

Comment:
Not sure If I have the right context here

Reply:
Thanks @kelbrown20. This is a great start. But for the supported pipelines with SDG we only have `simple` and `full`, eval and schema are sub parts for some pipelines if and when needed. So let us limit to just simple and full. If we were to go in depth we could talk about separate pipelines for freeform, grounded and knowledge for simple and full. WDYT?

Reply:
Today we only ship a "full" and a "simple" pipeline, although users can technically define their own pipelines as well. Exact hardware requirements for each is fluid, as with each pipeline there's a question of the models you use. Out of the box, the "full" pipeline is designed to run on a machine large enough to host the mixtral-8x7b-instruct model, which means approximately 120GB+ of GPU memory. The closest typical machine size to that is something like 8x24GB GPUs for 192GB total of GPU memory. However, you can run this pipeline with less - sometimes even much less if you instead use a quantized mixtral teacher model - so what exactly do we want to document for hardware requirements there?

The "simple" pipeline has much lower requirements - I'm not sure the exact minimum specs, but typical consumer hardware is plenty for it.